src/connector/src/source/cdc/source/reader.rs

```diff
@@ -202,14 +203,25 @@ impl<T: CdcSourceTypeTrait> CommonSplitReader for CdcSplitReader<T> {
         let metrics = self.source_ctx.metrics.clone();
 
         while let Some(result) = rx.recv().await {
-            let GetEventStreamResponse { events, .. } = result?;
-            tracing::trace!("receive {} cdc events ", events.len());
-            metrics
-                .connector_source_rows_received
-                .with_label_values(&[source_type.as_str_name(), &source_id])
-                .inc_by(events.len() as u64);
-            let msgs = events.into_iter().map(SourceMessage::from).collect_vec();
-            yield msgs;
+            match result {
+                Ok(GetEventStreamResponse { events, .. }) => {
+                    tracing::trace!("receive {} cdc events ", events.len());
+                    metrics
+                        .connector_source_rows_received
+                        .with_label_values(&[source_type.as_str_name(), &source_id])
+                        .inc_by(events.len() as u64);
+                    let msgs = events.into_iter().map(SourceMessage::from).collect_vec();
+                    yield msgs;
+                }
+                Err(e) => {
+                    GLOBAL_ERROR_METRICS.cdc_source_error.report([
+                        source_type.as_str_name().into(),
+                        source_id.clone(),
+                        e.to_string(),
```

Comment:
FYI it is not good practice to include detailed error message as metric label. Error message may contain the offending input value and could lead to [high cardinality](https://prometheus.io/docs/practices/naming/#labels). It shall just be an error type as label, which the details going into logs.